lib/puppetlabs_spec_helper/rake_tasks.rb

```diff
@@ -141,6 +142,15 @@ def clone_repo(scm, remote, target, ref=nil, branch=nil, flags = nil)
     fail "Unfortunately #{scm} is not supported yet"
   end
   result = system("#{scm} #{args.flatten.join ' '}")
+  unless dirpath.nil?
+    tmpdir = SecureRandom.hex
+    system('setopt -s dotglob')
```

Comment:
this will spawn a shell, setopt there, then destroy the shell. The setting then will not be available to the next commands.